a10_octavia/controller/worker/tasks/a10_database_tasks.py

```diff
@@ -114,14 +115,20 @@ def execute(self, loadbalancer):
 
 
 class GetVThunderByLoadBalancer(BaseDatabaseTask):
-
-    """ Get VThunder details from LoadBalancer"""
+    """Get VThunder from db using LoadBalancer"""
 
     def execute(self, loadbalancer):
         loadbalancer_id = loadbalancer.id
         vthunder = self.vthunder_repo.get_vthunder_from_lb(
             db_apis.get_session(), loadbalancer_id)
-        LOG.info("Successfully fetched vThunder details for LB")
+        if (vthunder and vthunder.undercloud and vthunder.hierarchical_multitenancy and
```

Comment:
```python
        create_amp_for_lb_subflow.add(a10_database_tasks.GetVThunderByLoadBalancer(
            name=sf_name + '-' + a10constants.VTHUNDER_BY_LB,
            requires=constants.LOADBALANCER,
            provides=a10constants.VTHUNDER))
```

This is supposed to provide the vthunder that was just created. If we fail get this we should fail and revert.

Reply:
added raise